docs/content/specs/bicep/_index.md

```diff
@@ -683,3 +665,62 @@ For reference, please refer to the following examples:
 ---
 
 <br>
+
+## Resource Module Requirements
+
+Listed below are both functional and non-functional requirements for Bicep [AVM Resource Modules](/Azure-Verified-Modules/specs/shared/module-classifications/).
+
+### Non-Functional Requirements (BCPRMNFR)
+
+{{< hint type=note >}}
+This section includes **resource module level, non-functional requirements (BCPRMNFR)** for Bicep.
+{{< /hint >}}
+
+---
+
+<br>
+
+#### ID: BCPRMNFR1 - Category: Testing - Expected Test Directories
+
+Module owners **MUST** create the `defaults`, `waf-aligned` folders within their `/tests/e2e/` directory in their resource module source code and **SHOULD** create a `max` folder also. Module owners **CAN** create additional folders as required. Each folder will be used as described for various test cases.
+
+##### Defaults tests (**MUST**)
+
+The `defaults` folder contains a test instance that deploys the module with the minimum set of required parameters.
+
+This includes input parameters of type `Required` plus input parameters of type `Conditional` marked as required for WAF compliance.
+
+This instance has heavy reliance on the default values for other input parameters. Parameters of type `Optional` **SHOULD NOT** be used.
+
+##### WAF aligned tests (**MUST**)
+
+The `waf-aligned` folder contains a test instance that deploys the module in alignment with the best-practices of the Azure Well-Architected Framework.
+
+This includes input parameters of type `Required`, parameters of type `Conditional` marked as required for WAF compliance, and parameters of type `Optional` useful for WAF compliance.
+
+Parameters and dependencies which are not needed for WAF compliance, **SHOULD NOT** be included.
+
+##### Max tests (**SHOULD**)
```

Comment:
Careful, controversal opinon: I'm a bit torn to ask contributors to implement a max test. The reason we added those to begin with was to have an easier time testing many parameters at once without deploying too many instances. Yet, as we've already seen, other contributors prefer to rather have the same parameters spread across multiple different test files (e.g. Cosmos DB). 

Maybe we can find a better way to package the message 'you should implement as many tests as you see fit to test as many different test scenarios you need'.

Reply:
Right, I'm just rethinking it as this is how contributors 'lived' it (in some cases) and I hence think it would probably make sense to rather ask for validating as many parameters as possible, through as many test files as it takes. We'd recommend a `max` test to make it easy to maintain, but if one splits the same tests across 5 different files (like one contributor did to make it easier to troubleshoot), so be it. Am I making sense? 😄

Reply:
It does. Isn't a should enough? How about mentioning something like this in a note under the should max section: you can avoid a max test if you already covered the whole parameter validation in use case specific tests

Reply:
I guess that would also work. My suggestion is, I guess, a more fundamental change in how the tests are 'framed'. Just wanted to bring it up as it has been bothering me for a while and you happened to open a PR on that topic 😄